### PR TITLE
Bugfix/cwd flag  closes #1160

### LINKED ISSUE #1160: Bug when --cwd --folder and --script are given.
## Describe the bug

The logic in calculating relative paths appears to be flawed.

## To reproduce

Given a tree structure
```
project_root
        ├── notebooks
        │   ├── run_clearml.py
```

```bash
clearml-task --project testing --name test_run --script notebooks/run_clearml.py --task-type testing  --folder $PWD --queue default --cwd notebooks
```

## Expected behaviour
Error: 'notebooks' does not start with  $PWD...

## Environment
* Ubuntu 22.04
* ClearML SDK Version
* clearml   1.13.2
* python 3.10.12
* OS Linux
## Related Discussion
If this continues a slack thread, please provide a link to the original slack thread.

### PR DESCRIPTION
**Pull request by dimidagd**
closes #1160 # A bug was preventing the use of --cwd and --folder. More specifically a function call to Path(path1).relative_to(path2) , where path1 has been stripped of the trailing directories, resulted in a specific exception being raised.

### How to test fix
Given a tree structure
```
project_root
        ├── notebooks
        │   ├── run_clearml.py
```

Running 
```bash
clearml-task --project testing --name test_run --script notebooks/run_clearml.py --task-type testing  --folder $PWD --queue default --cwd notebooks
```

Expected behaviour

```
Creating new task
Repository Detected
{
  "repository": "###",
  "version_num": "5946cd78e5a69e0a6916af8fb734c907de5feace",
  "branch": "main",
  "working_dir": "notebooks",
  "entry_point": "run_clearml.py"
}

New task created id=9a140105541a40478b8e0dac87317d2b

```

### CODE REVIEW

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @dimidagd ! I think this is a good change, but can you please make some changes such that the section you changed is a bit clearer? It's a bit hard to follow. You could add some comments that exemplifies what is going on in this section

**Comment by jkhenning** [comment]
@dimidagd any update on this?

**Comment by eugen-ajechiloae-clearml** [comment]
@dimidagd yes, the section itself (even excluding the changes) is a bit hard to read. Can you please add a few comments that could make it a bit easier to follow?

File: clearml/backend_interface/task/populate.py [hunk 1]
```diff
@@ -218,15 +215,14 @@ def create_task(self, dry_run=False):
...
-                            Path(repo_info.script['repo_root']) / self.cwd).as_posix()
+                cwd = (     Path(repo_info.script['repo_root']) / self.cwd).as_posix()
                 if not Path(cwd).is_dir():
                     raise ValueError("Working directory \'{}\' could not be found".format(cwd))
-                cwd = Path(cwd).relative_to(repo_info.script['repo_root']).as_posix()
                 entry_point = \
                     Path(repo_info.script['repo_root']) / repo_info.script['working_dir'] / repo_info.script[
                         'entry_point']
                 entry_point = entry_point.relative_to(cwd).as_posix()
+                cwd = Path(cwd).relative_to(repo_info.script['repo_root']).as_posix()
```
Comment (dimidagd) [inline clearml/backend_interface/task/populate.py:225]: Relative path to *cwd* from *entry_points* can be now resolved, as *cwd* is full path and not stripped.

File: clearml/backend_interface/task/populate.py [hunk 2]
```diff
@@ -218,15 +215,14 @@ def create_task(self, dry_run=False):
             task_state['script']['requirements'] = repo_info.script.get('requirements') or {}
             if self.cwd:
                 self.cwd = self.cwd
-                cwd = self.cwd if Path(self.cwd).is_dir() else (
-                            Path(repo_info.script['repo_root']) / self.cwd).as_posix()
+                cwd = (     Path(repo_info.script['repo_root']) / self.cwd).as_posix()
                 if not Path(cwd).is_dir():
                     raise ValueError("Working directory \'{}\' could not be found".format(cwd))
-                cwd = Path(cwd).relative_to(repo_info.script['repo_root']).as_posix()
                 entry_point = \
```
Comment (dimidagd) [inline clearml/backend_interface/task/populate.py:221]: Entry point is always full

File: clearml/backend_interface/task/populate.py [hunk 3]
```diff
@@ -218,15 +215,14 @@ def create_task(self, dry_run=False):
             task_state['script']['requirements'] = repo_info.script.get('requirements') or {}
             if self.cwd:
                 self.cwd = self.cwd
-                cwd = self.cwd if Path(self.cwd).is_dir() else (
-                            Path(repo_info.script['repo_root']) / self.cwd).as_posix()
+                cwd = (     Path(repo_info.script['repo_root']) / self.cwd).as_posix()
                 if not Path(cwd).is_dir():
                     raise ValueError("Working directory \'{}\' could not be found".format(cwd))
-                cwd = Path(cwd).relative_to(repo_info.script['repo_root']).as_posix()
```
Comment (dimidagd) [inline clearml/backend_interface/task/populate.py:225]: move *cwd* override by relative path later on, because *entry_point* has to be resolved

**Comment by dimidagd** [comment]
> @dimidagd yes, the section itself (even excluding the changes) is a bit hard to read. Can you please add a few comments that could make it a bit easier to follow?

Have added comments now.

**Comment by jkhenning** [comment]
@dimidagd it seems there are conflicts - can you please resolve them?
